src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -570,6 +574,50 @@ protected void validateSequenceDictionaries() {
         }
     }
 
+    /**
+     * Register the reference file (and associated dictionary and index) to be downloaded to every node using Spark's
+     * copying mechanism ({@code SparkContext#addFile()}).
+     * @param ctx the Spark context
+     * @param referenceFile the reference file, can be a local file or a remote path
+     * @return the reference file name; the absolute path of the file can be found by a Spark task using {@code SparkFiles#get()}
+     */
+    protected String addReferenceFilesForSpark(JavaSparkContext ctx, String referenceFile) {
+        Path referencePath = IOUtils.getPath(referenceFile);
+        Path indexPath = ReferenceSequenceFileFactory.getFastaIndexFileName(referencePath);
+        Path dictPath = ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(referencePath);
+
+        // TODO: check existence of all files
+        ctx.addFile(referenceFile);
+        ctx.addFile(indexPath.toUri().toString());
+        ctx.addFile(dictPath.toUri().toString());
+
+        return referencePath.getFileName().toString();
+    }
+
+    /**
+     * Register the VCF file (and associated index) to be downloaded to every node using Spark's copying mechanism
+     * ({@code SparkContext#addFile()}).
+     * @param ctx the Spark context
+     * @param knownVariants the known variant (VCF) files, can be local files or remote paths
+     * @return the reference file name; the absolute path of the file can be found by a Spark task using {@code SparkFiles#get()}
+     */
+    protected List<String> addKnownSitesForSpark(JavaSparkContext ctx, List<String> knownVariants) {
+        for (String vcfFileName : knownVariants) {
+            String vcfIndexFileName;
+            if (vcfFileName.endsWith(IOUtil.VCF_FILE_EXTENSION)) {
+                vcfIndexFileName = vcfFileName + IOUtil.VCF_INDEX_EXTENSION;
+            } else if (vcfFileName.endsWith(IOUtil.COMPRESSED_VCF_FILE_EXTENSION)) {
+                vcfIndexFileName = vcfFileName + IOUtil.COMPRESSED_VCF_INDEX_EXTENSION;
+            } else {
+                throw new IllegalArgumentException("Unrecognized known sites file extension. Must be .vcf or .vcf.gz");
+            }
+            // TODO: check existence of all files
+            ctx.addFile(vcfFileName);
```

Comment:
When these files are added, a copy gets created on each worker node? So if a single machine houses 4 worker nodes then we would expect it to house 4 copies of the file?